libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -674,5 +829,580 @@ private void RestoreParentGenerator(ITurnContext context)
                 context.TurnState.Remove(this.generatorTurnKey);
             }
         }
+
+        private void ProcessEntities(SequenceContext context)
+        {
+            if (Schema != null)
+            {
+                var queues = EventQueues.Read(context);
+                var entities = NormalizeEntities(context);
+                var utterance = context.Context.Activity?.AsMessageActivity()?.Text;
+                if (!context.GetState().TryGetValue<string[]>(DialogPath.ExpectedProperties, out var expected))
+                {
+                    expected = new string[0];
+                }
+
+                entities["utterance"] = new List<EntityInfo> { new EntityInfo { Priority = int.MaxValue, Coverage = 1.0, Start = 0, End = utterance.Length, Name = "utterance", Score = 0.0, Type = "string", Value = utterance, Text = utterance } };
+                var updated = UpdateLastEvent(context, queues, entities, expected);
+                var newQueues = new EventQueues();
+                var recognized = AssignEntities(entities, expected, newQueues);
+                var unrecognized = SplitUtterance(utterance, recognized);
+                recognized.AddRange(updated);
+
+                // TODO: Is this actually useful information?
+                context.GetState().SetValue(TurnPath.UNRECOGNIZEDTEXT, unrecognized);
+                context.GetState().SetValue(TurnPath.RECOGNIZEDENTITIES, recognized);
+                queues.Merge(newQueues);
+                var turn = context.GetState().GetValue<uint>(DialogPath.EventCounter);
+                CombineOldEntityToPropertys(queues, turn);
+                queues.Write(context);
+            }
+        }
+
+        private List<string> SplitUtterance(string utterance, List<EntityInfo> recognized)
```

Comment:
>SplitUtterance [](start = 29, length = 14)

is this really splitting?  Can't you have entities which overlap or are the same start range?